packages/ra-tree-ui-materialui/src/TreeNode.tsx

```diff
@@ -0,0 +1,538 @@
+import React, {
```

Comment:
this file needs to be split into several smaller components, it's too hard to follow